lib/contentful_middleman/commands/contentful.rb

```diff
@@ -0,0 +1,117 @@
+require 'middleman-core/cli'
+require 'middleman-blog/uri_templates'
+require 'date'
+require 'digest'
+require 'contentful_middleman/commands/context'
+require 'contentful_middleman/tools/backup'
+
+module Middleman
+  module Cli
+    # This class provides an "contentful" command for the middleman CLI.
+
+    class VersionHash
+      def self.read_for_space(space_name)
+        hashfilename_for_space = hashfilename(space_name)
+        ::File.read(hashfilename_for_space) if File.exist? hashfilename_for_space
+      end
+
+      def self.write_for_space_with_entries(space_name, entries)
+        sorted_entries           = entries.sort {|a, b| a.id <=> b.id}
+        ids_and_revisions_string = sorted_entries.map {|e| "#{e.id}#{e.revision}"}.join
+        entries_hash             = Digest::SHA1.hexdigest( ids_and_revisions_string )
+
+        File.open(hashfilename(space_name), 'w') { |file| file.write(entries_hash) }
+
+        entries_hash
+      end
+
+      def self.hashfilename(space_name)
+        ::File.join(Contentful.source_root, ".#{space_name}-space-hash")
+      end
+    end
+
+
+    class Contentful < Thor
+      include Thor::Actions
+      include ContentfulMiddleman::Tools::Backup::InstanceMethods
+
+      check_unknown_options!
+
+      namespace :contentful
+      desc 'contentful', 'Import data from Contentful'
+
+      method_option "rebuild",
+        aliases: "-r",
+        desc: "Rebuilds the site if there were changes in the imported data"
+
+      def self.source_root
+        ENV['MM_ROOT']
+      end
+
+      # Tell Thor to exit with a nonzero exit code on failure
+      def self.exit_on_failure?
+        true
+      end
+
+      def contentful
```

Comment:
Fixed in https://github.com/madtrick/contentful_middleman/commit/2a2518156f5545a2972935906f2533e5e45371f0